src/app/components/cohort-charts/chart.service.spec.js

```diff
@@ -268,4 +278,94 @@ describe('ChartService Unit Tests', function () {
+
+        var chart = {
+            tsLabel: {
+                filters: []
+            },
+            data: function () {
+                return [1, 2];
+            }
+        }
+
+        it('should add "pie-legend-bold" class to text tags with filters and "dc-legend-item" tags', function () {
+            chart.tsLabel.filters = ['female'];
+            var items = ChartService.emphasizeChartLegend(chart, divWithDcItems);
+            expect(items.length).toEqual(1);
+            expect(items[0].hasClass('pie-legend-bold')).toBe(true);
+        });
+
+        it('should not add "pie-legend-bold" class to text tags without filters or "dc-legend-item" tags', function () {
+            chart.tsLabel.filters = [];
+            var items = ChartService.emphasizeChartLegend(chart, divWithDcItems);
+            expect(items.length).toEqual(0);
+
+            chart.tsLabel.filters = ['male'];
+            items = ChartService.emphasizeChartLegend(chart, divWithoutDcItems);
+            expect(items.length).toEqual(0);
+        });
+
+    });
+
+    describe('createCohortChart', function () {
+        var label, el, subjects;
+
+        beforeEach(function () {
+            label = {
+                $$hashKey: "object:306",
+                col: 0,
+                row: 0,
+                filter: undefined,
+                label: "/Public Studies/AAA_TRANSLOCATION/Gender/",
+                labelId: 0,
+                name: "Gender",
+                resolved: false,
+                sizeX: 3,
+                sizeY: 3,
+                type: "string"
+            };
+            el = document.createElement('div');
+            subjects = [{
+                id: 1, gender: 'male', labels: {}
+            }, {
+                id: 2, gender: 'female', labels: {}
+            }, {
+                id: 3, gender: 'male', labels: {}
+            }, {
+                id: 4, gender: 'female', labels: {}
+            }, {
+                id: 5, gender: 'unknown', labels: {}
+            }];
+            ChartService.reset();
+            ChartService.cs.subjects = subjects;
+            ChartService.cs.crossfilter = crossfilter(subjects);
+
+        });
+
+        it('should listen to the filtered event when filtered', function () {
+            spyOn(ChartService, 'createCohortChart').and.callThrough();
+            spyOn(DcChartsService, 'getPieChart').and.callThrough();
+            
+            var _chart = ChartService.createCohortChart(label, el);
+            /*
+             * renderlet called each time a chart is created
+             * postRender is called after the renderlet event
+             */
+            _chart.on('postRender', function () {
```

Comment:
using $scope.$digest does not seem to delay the 'expect(ChartService.emphasizeChartLegend).toHaveBeenCalled();' clause to be executed, I keep the initial code where the postRender lis tener is called and check expectancy there. Also, because postRender is called after renderlet, we can guarantee that that emphasizeChartLegend has been called when the postRender is invoked.